### ISSUE TITLE
slow download

### ISSUE THREAD

**Issue by pdudnik** [GoogleCloudPlatform/google-cloud-python#2927]
I am trying to download a 400M gcs file. I am using https://github.com/GoogleCloudPlatform/google-cloud-python/blob/ce6756fbe3633c74fd742567654565147628f4ba/storage/google/cloud/storage/blob.py. I noticed that by default my download looks to be chunked due to this setting here 

https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/core/google/cloud/streaming/transfer.py#L46

As a result downloading the file results in 400 calls to GCS which significantly slows down the download.

Is there some clean way I can override that when using blob.download_to_file?

**Comment by dhermes** [comment]
Thanks for reporting. This is actually deeper than it seems. The "correct" fix is for u to remove the (non-public) `google.streaming` stuff that this relies on and get a better chunking story (that doesn't rely on `httplib2`).

For a fix that works right now, you can duplicate [the source][1] but pass [`chunksize`][2] to `Download`. Also, `gsutil` (the CLI tool) has a **very** optimized strategy for fast downloads.

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/cf249243dc029aa1b2ab23bde780843ee29c3df1/storage/google/cloud/storage/blob.py#L325-L346
[2]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/cf249243dc029aa1b2ab23bde780843ee29c3df1/core/google/cloud/streaming/transfer.py#L75

**Comment by bjwatson** [comment]
@lukesneeringer says this is blocked on httplib2 work.

**Comment by lukesneeringer** [comment]
The _correct_ solution is blocked on #1998.

What would be the benefits and drawbacks of increasing the default, though? Would it be okay to make the default chunk size 10 MB instead of 1 MB?

**Comment by pdudnik** [comment]
I think gcs has a way of Auto detecting what's the best chunk size, so I'd
ask them. I know they have solutions for this issue

**Comment by evanj** [comment]
I think this is basically a duplicate of #2222

**Comment by lukesneeringer** [comment]
@dhermes Is this easier now that #1998 is done?

**Comment by dhermes** [comment]
Not easier or harder. AFAIK there is no perfect magic chunking answer, @thobrla has said before that downloading in a single request (vs. chunks) is almost always the right answer

**Comment by tseaver** [comment]
Duplicate:  #2222.
